docs(world): add tables to the new docs (#1830)
latticexyz/mud: next-docs/pages/world/tables.mdx

# Tables

`World` uses the [store tables](/store/tables), but adds [access control](/world/namespaces-access-control).
For onchain tables, the data is stored by the `World` contract, [which is also a `StoreData`](https://github.com/latticexyz/mud/blob/main/packages/world/src/World.sol#L43).

> When a `System` attempts to write to storage, the request goes into [`StoreSwitch`](https://github.com/latticexyz/mud/blob/main/packages/store/src/StoreSwitch.sol). This library decides which approach to use:

Reviewer: ```suggestion
When a `System` reads or writes storage via [table libraries](/store/table-libraries), the request goes into [`StoreSwitch`](https://github.com/latticexyz/mud/blob/main/packages/store/src/StoreSwitch.sol). This library decides which approach to use:
```

- If the `System` is in the root namespace, then it was called with [`delegatecall`](https://docs.soliditylang.org/en/v0.8.22/introduction-to-smart-contracts.html#delegatecall-and-libraries).

>   This means it has the same storage as the `World` and can write directly to storage using [`StoreCore`](https://github.com/latticexyz/mud/blob/main/packages/store/src/StoreCore.sol).

Reviewer: ```suggestion
  This means it inherits the `World` storage and can write directly to storage using [`StoreCore`](https://github.com/latticexyz/mud/blob/main/packages/store/src/StoreCore.sol).
```

  These calls bypass access control.

- If the `System` is in any other namespace, then it was called with [`call`](https://docs.soliditylang.org/en/v0.8.22/introduction-to-smart-contracts.html#message-calls) and has to call back into the `World` using [`IStore`](https://github.com/latticexyz/mud/blob/main/packages/store/src/IStore.sol).
  These calls go through access control.
  They are only permitted if the `System` has access to the table in question.

>   By default a `System` has access to its own namespace and therefore to all the tables inside it - but the namespace owner can modify that if necessary.

Reviewer: ```suggestion
  By default a `System` has access to its own namespace and therefore to all the tables inside it.
  Additional [access can be granted](./namespaces-access-control) by the namespace owner.
```

> ![Interaction between the `World`, a `System`, and a table](./world-table.svg)

Reviewer: This part feels similar to the "detailed description" in the systems page. We could extract it to a separate `.mdx` file, ignore that `.mdx` file in the menu, and import here and on the systems page (to keep aligned and only have to change it in once place)

> 1. An external entity (contract or externally owned account) calls on the `World` a function called `myns_sys_fun`.

Reviewer: Same feedback here as in the system doc: let's try to avoid abbreviations whenever possible
```suggestion
1. An account calls a function called `namespace_system_function` via the `World`.
```
